accepted/00000-ptf-repos-in-uyuni.md

```diff
@@ -0,0 +1,104 @@
+- Feature Name: PTF repositories in Uyuni
+- Start Date: 2021-05-17
+
+# Summary
+[summary]: #summary
+
+SUSE want to provide customers PTFs (Product Temporary Fixes) in repositories.
+Uyuni should be able to offer these repositories as channels for syncing and management.
+
+# Motivation
+[motivation]: #motivation
+
+PTFs are Product Temporary Fixes for Service Requests opened by SUSE customers.
+The resulting packages are the last released maintenance update plus a patch which fixes the actual issue.
+
+PTFs are made for dedicated customers. This means every customer has different PTF repositories, with different content and fixes for different service requests.
+
+With Uyuni we should provide the possibility to mirror the content of these repositories and let the customer use it for there clients.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The mapping of SCC repositories to Uyuni Channels is done via a prepared product tree file which contains the definitions for all products and its repositories.
+It enhances the information from SCC with values required for Uyuni.
+
+Such an entry is defined by 3 values:
+
+- the scc product id
+- the scc product id of the root product
+- the scc repository id
+
+The following values are set by the product tree:
+
+- channel label: it must be unique on the whole Uyuni Server
+- parent channel label: what is the parent of this channel
+- channel name: similar to channel label but human readable
+- update tag: a special value to modify the errata id during import
+- mandatory: defines if this channel is required for the product it belongs to
+
+As PTF repositories are customer specific, such a hard coded definition cannot 
+be provided upfront.
+They must be generated out of existing data.
+
+As SCC needs to crawl the buildservice to find all the existing PTF repositories, they agreed on a path structure to map the repositories to the products.
+The defined path is:
+
+    PTF/Release/<Customer Account ID>/<Product Identifier>/<Product Version>/<Product Architecture>/ptf[_debug]/
+
+When Uyuni is listing the repositories available for an organizations, it may find repositories which `id` is not known in the complete product tree.
+To identify them as PTF repositories we can check if the path start with `PTF/Release`.
+
+If they are PTF repositories, we need to generate the missing values for the Channel mapping like this:
+
+- scc repository id: take it from the repository definition
+- scc product id: search for the product using `Product Identifier`, `Product Version`, and `Product Architecture` like specified in the URL
+- scc root product id: find all existing root products for the product id. We need to generate an entry for all of them
+
+The values for update tag and mandatory can be hard coded as they are the same 
+for all PTF repositories.
+
+- update tag: empty
```

Comment:
Are we assuming the same patch ID delivered as PTF will be later on delivered by the officially released update for all customers? Will the errata entry in Uyuni be maintained in this case, and packages "re-linked" from the PTF channel to the corresponding Update channel?

Reply:
PTFs will not get "Patches". They work package based only. Another reason why we do not need to care about the update tag. I will elaborate more about it in a new section added to this RFC